src/ACEtk/details/verify/Positive/src/ctor.hpp

```diff
@@ -0,0 +1,35 @@
+template< typename Arg = std::initializer_list< typename Range::value_type >,
+          utility::Require< false, IsPositive, Arg > = true,
+          utility::Require< true, utility::is_range, Arg > = true,
+          utility::Require< true, std::is_constructible, Range, Arg > = true
+        >
+Positive( Arg&& arg ) :
+  Range( check( std::forward< Arg >( arg ) ) )
+{ }
+
+template< typename... Args, 
+          utility::Require< true, std::is_constructible, Range, Args... > = true
+        >
+Positive( Args&&... args ) :
+  Range( std::forward< Args >( args )... )
+{ 
+  check( *this );
+}
+
+template< typename Arg = std::initializer_list< typename Range::value_type >,
+          utility::Require< false, IsPositive, Arg > = true,
+          utility::Require< true, IsSorted, Arg > = true,
+          utility::Require< true, utility::is_range, Arg > = true,
+          utility::Require< true, std::is_constructible, Range, Arg > = true
+        >
+Positive( Sorted< Arg >&& arg ) :
```

Comment:
The `Sorted` template here is pessimizing. You've enforced this with a requirement.

Reply:
In fact this can removed all together. You're check function has the appropriate logic to handle the sorted case.